File: 백준/Bronze/15829. Hashing/Hashing.js
const fs = require("fs");
const input = fs.readFileSync("/dev/stdin").toString().trim().split("\n");

const [l, s] = input;
const M = 1234567891;
const R = 31;
let answer = 0;
let rPower = 1;

for (let i = 0; i < Number(l); i++) {
  const a = s[i].charCodeAt() - "a".charCodeAt() + 1;
  answer = (answer + ((a * rPower) % M)) % M;
  rPower = (rPower * R) % M;
}

console.log(answer);

File: 백준/Bronze/15829. Hashing/Hashing.py
import sys

input = sys.stdin.readline

l, s = [input() for _ in range(2)]
M = 1234567891
R = 31
answer = 0
rPower = 1

for i in range(int(l)):
  a = ord(s[i]) - ord("a") + 1
  answer = (answer + ((a * rPower) % M)) % M
  rPower = (rPower * R) % M

print(answer)
